### ISSUE TITLE
[BUG] Failure during codegen

### ISSUE THREAD

**Issue by Fusioon** [beefytech/Beef#1561]
Code to reproduce:
```
class Program
{
    struct Vec
    {
        float a,b,c,d,e,f,g,h;
    }

    static Vec _field = default;
    static readonly ref Vec X
    {
        get
        {
            return ref _field; 
        }
    }

    // Works
    static readonly ref Vec Foo()
    {
        return ref X;
    }

    // Doesn't work :(
    //static readonly ref int Bar() => ref X;

    public static void Main()
    {
        int q = 0;
        let qwe = Foo();
        int w = 0;
        Console.WriteLine((void*)(&w - &q));
    }
}
```
Above code fails to compile with following error: `Module 'Tests_Program' failed during codegen with: Failure during codegen of ?Main@Program@Tests@bf@@SAXXZ: VRegAddr used without ForceMem`

Also there is a bug where using expression body definition with `readonly ref` doesn't work.  uncomment `static readonly ref int Bar() => ref X;` for more info

Tested with:  Nightly 05/10/2022

**Comment by Fusioon** [comment]
I have also noticed that there doesn't seem to be a way to assign `readonly ref` to local variable, is this by design or bug?

**Comment by Fusioon** [comment]
Oh nevermind seems like `let` can be used for readonly ref variables.

**Comment by bfiete** [comment]
Fixed at 5bd8aac41b0a8d68600581a8ed410680a99dcb9b

What was the issue with readonly ref local variables? This worked: `readonly ref Vec local = ref Foo();`, for example.

**Comment by Fusioon** [comment]
I forgot to use `ref` before the `Foo` call
